django/db/models/base.py

```diff
@@ -450,6 +450,14 @@ def __init__(self, db=None):
         # Necessary for correct validation of new instances of objects with explicit (non-auto) PKs.
         # This impacts validation only; it has no effect on the actual save.
         self.adding = True
+        self.cache = {}
+
+    def set_fields_cache(self, field, value):
```

Comment:
Agree, now that we have the mixin I think it's much better to let that handle the logic with setting field cache.
Regarding `get_fields_cache()`, the reason I thought to leave it in is because as we discussed it's possible to use this cache for other things that are not fields and so I figured it's best to provide a method that gets all fields and returns the dictionary with only field->value vs anything that's not a field. I don't feel too strongly about this though, so I'll go ahead and remove.